src/dynamic_boundary_conditions/tide/sea_level_rise_data.py

```diff
@@ -145,28 +95,70 @@ def read_slr_data_from_files(slr_data_dir: pathlib.Path) -> gpd.GeoDataFrame:
-    geometry = gpd.points_from_xy(slr_nz['lon'], slr_nz['lat'], crs=4326)
-    slr_nz_with_geom = gpd.GeoDataFrame(slr_nz, geometry=geometry)
-    # Convert all column names to lowercase
-    slr_nz_with_geom.columns = slr_nz_with_geom.columns.str.lower()
+
+    # Create dictionary to store the sea level rise dataset
+    slr_nz_dict = {}
+    for dataset_path in slr_data_dir.glob("*.csv"):
+        name_dataset = dataset_path.stem
+        slr_nz_dict[name_dataset] = pd.read_csv(dataset_path)
+
+    # Column name used to merge
+    left_column_name = 'Site ID'
+    right_column_name = 'site'
+    # Merge Site Details dataframe and Sea level projections tables WITH VLM
+    slr_nz_merge_vlm = pd.merge(
+        slr_nz_dict["NZ_VLM_final_May24"], slr_nz_dict["NZSeaRise_proj_vlm"],
+        left_on=left_column_name, right_on=right_column_name,
+        how='left', validate='1:m',
+        suffixes=('_y', '')
+    )
+    slr_nz_merge_vlm['add_vlm'] = True
+
+    # Merge Site Details dataframe and Sea level projections tables WITHOUT VLM
+    slr_nz_merge_novlm = pd.merge(
+        slr_nz_dict["NZ_VLM_final_May24"], slr_nz_dict["NZSeaRise_proj_novlm"],
+        left_on=left_column_name, right_on=right_column_name,
+        how='left', validate='1:m',
+        suffixes=('_y', '')
+    )
+    slr_nz_merge_novlm['add_vlm'] = False
+
+    # Concatenate all the dataframes in the list
+    slr_nz = pd.concat([slr_nz_merge_vlm, slr_nz_merge_novlm], axis=0).reset_index(drop=True)
+
+    # Remove unnamed columns
+    slr_nz_df = slr_nz.loc[:, ~slr_nz.columns.str.contains('^Unnamed')]
+    # Remove site column
+    slr_nz_df = slr_nz_df.drop(columns=['site'])
+    # Rename the columns
+    slr_nz_df = slr_nz_df.rename(columns={
+        left_column_name: 'siteid',
+        'Lon': 'lon',
+        'Lat': 'lat',
+        'Vertical Rate (mm/yr)': 'vertical_rate',
+        'Vertical Rate - BOP corrected (mm/yr)': 'vertical_rate_bop',
+        '1-sigma uncertainty (mm/yr)': 'sigma_uncertainty',
+        'Number of obs': 'number_of_obs',
+        'Quality Factor': 'quality_factor',
+        'Average distance between coastal point and observations': "average_distance",
+        'Confidence': 'confidence_level',
+        '0.17': 'p17',
+        '0.5': 'p50',
+        '0.83': 'p83',
+        'SSP': 'ssp'
+    })
+
+    # Remove '_confidence' in confidence_level column
+    slr_nz_df['confidence_level'] = slr_nz_df['confidence_level'].str.split('_').str[0]
+    # Merge SSP and scenario into one column 'ssp_scenario'
+    slr_nz_df['ssp_scenario'] = slr_nz_df['ssp'].astype(str) + '-' + slr_nz_df['scenario'].astype(str)
+    # Remove 'ssp' and 'scenario' column
+    slr_nz_df = slr_nz_df.drop(columns=['ssp', 'scenario'])
+
+    # Add geometry
+    geometry = gpd.points_from_xy(slr_nz_df['lon'], slr_nz_df['lat'], crs=4326)
+    slr_nz_with_geom = gpd.GeoDataFrame(slr_nz_df, geometry=geometry)
+
     return slr_nz_with_geom
```

Comment:
The `slr_nz_with_geom` GeoDataFrame contains 2,092,288 rows and 17 columns, while the `NZSeaRise_proj_novlm.csv` and `NZSeaRise_proj_vlm.csv` files each have 1,046,912 records, totalling 2,093,824 records. Upon review, it appears that some site IDs are missing in the `NZ_VLM_final_May24.csv` file, specifically IDs 7264, 7274, 7328, 7347, 7365, and 7421. Although we might not be able to address the missing data, it's important to be aware of this issue.

Reply:
Good spotting Casey, do you know if these IDs are for sites that are of significance to us?

Reply:
> Good spotting Casey, do you know if these IDs are for sites that are of significance to us?

Since those site IDs are missing, the latitude and longitude for them are also unavailable. However, they might be located near the neighbouring site IDs. We'll need to convert the .csv file into a geometry file and view it in QGIS to investigate further.

Reply:
@caseyli92 I think we can probably ignore it, but if you'd like to investigate further then we can do that

Reply:
@LukeParky I think we can probably disregard this issue. We can’t access the missing data since it's not available, and if there is a nearby site, it will use data from that site instead.